fs/accounting/stats.go

```diff
@@ -624,17 +624,28 @@ func (s *StatsInfo) RemoveTransfer(transfer *Transfer) {
 	s.mu.Unlock()
 }
 
-// PruneTransfers makes sure there aren't too many old transfers
-func (s *StatsInfo) PruneTransfers() {
+// pruneTransfers removes transfers that already finished
+//
+// Must be called with the lock held
+func (s *StatsInfo) pruneTransfers() {
+	for i, tr := range s.startedTransfers {
+		if tr.IsDone() {
+			s.removeTransfer(tr, i)
+			break
+		}
+	}
+}
+
+// PruneTransfers makes sure there aren't too many old transfers.
+// It purges transfers if number of startedTransfers exceeds `limit`.
+func (s *StatsInfo) PruneTransfers(limit int) {
+	if limit < 0 {
+		return
+	}
 	s.mu.Lock()
 	// remove a transfer from the start if we are over quota
-	if len(s.startedTransfers) > maxCompletedTransfers+fs.Config.Transfers {
-		for i, tr := range s.startedTransfers {
-			if tr.IsDone() {
-				s.removeTransfer(tr, i)
-				break
-			}
-		}
+	if len(s.startedTransfers) > limit+fs.Config.Transfers {
```

Comment:
Wouldn't it be better to use `max(limit, fs.Config.Transfers)` here?

cc: @ncw